src/core/sock/sockinfo_tcp.cpp

```diff
@@ -1372,6 +1384,13 @@ err_t sockinfo_tcp::ip_output(struct pbuf *p, struct tcp_seg *seg, void *v_p_con
     int count = 0;
     void *cur_end;
 
+    if (unlikely(flags & XLIO_TX_PACKET_REXMIT)) {
+        const mem_buf_desc_t *mem_buf_desc = (const mem_buf_desc_t *)p;
+        if (unlikely(is_socket_in_error_state(mem_buf_desc, (struct tcp_pcb *)v_p_conn))) {
```

Comment:
same

Reply:
What do you think about making this method a member and doing:
p_si_tcp->is_socket_in_error_state()
Also the name is_socket_in_error_state is a bit confusing because it suggests some read only check but in fact it sets flags. Maybe check_previous_buffer_failure?